### PR TITLE
chore(tools): add check flag to `browser_compat.ts`

### PR DESCRIPTION
**Pull request by timreichen**
- adds check flag to the `browser_compat.ts` tool to check missing comments in file. It will inject missing comments if the flag is not present.
This feature is borrowed from the `check_licence.ts` tool.

### CODE REVIEW

File: _tools/check_browser_compat.ts [hunk 1]
```diff
@@ -26,24 +29,24 @@ function isBrowserCompatible(filePath: string): boolean {
...
+  if (isBrowserCompatible(path) && !hasBrowserCompatibleComment(source)) {
+    if (CHECK) {
+      console.log(
+        `${path} is likely browser-compatible and can have the "${DECLARATION}" comment added.`,
+      );
+    } else {
+      const index = source.indexOf(COPYRIGHT);
+      await Deno.writeTextFile(
+        path,
+        source.slice(0, index + COPYRIGHT.length) + "\n" + DECLARATION + "\n" +
```
Comment (kt3k) [inline _tools/check_browser_compat.ts:47]: The second `"\n"` looks redundant to me.

File: _tools/check_browser_compat.ts [hunk 2]
```diff
@@ -6,11 +6,14 @@
...
  * Run using: deno run --allow-read --allow-run _tools/check_browser_compat.ts
  */
-import { walkSync } from "../fs/walk.ts";
+
+import { walk } from "../fs/walk.ts";
+import { COPYRIGHT } from "./check_licence.ts";
 
 const ROOT = new URL("../", import.meta.url);
 const SKIP = [/(test|bench|\/_|\\_|testdata|version.ts)/];
 const DECLARATION = "// This module is browser compatible.";
```
Comment (kt3k) [inline _tools/check_browser_compat.ts:15]: Not directly related to this PR, but some file (`data_structures/comparators.ts`) seems having `/** This module is browser compatible. */` comment

File: _tools/check_browser_compat.ts (hunk 1 above)
Comment (iuioiua) [reply _tools/check_browser_compat.ts:47]: This section makes me wonder whether we should integrate both copyright and web compatibility checks. Either way, an idea for another issue/PR.

**Review by iuioiua — COMMENTED** [review]
Great idea!

File: _tools/check_browser_compat.ts (hunk 1 above)
Comment (timreichen) [reply _tools/check_browser_compat.ts]: > The second `"\n"` looks redundant to me.

removed

File: _tools/check_browser_compat.ts (hunk 2 above)
Comment (timreichen) [reply _tools/check_browser_compat.ts:15]: Ok, I guess we should replace that with a single line comment in another PR.

File: _tools/check_browser_compat.ts (hunk 1 above)
Comment (timreichen) [reply _tools/check_browser_compat.ts]: > This section makes me wonder whether we should integrate both copyright and web compatibility checks. Either way, an idea for another issue/PR.

I agree, this looks ugly and hacky. I had a try with the ts compiler API which would work without direct string manipulation, but the ts printer doesn't retain format and whitespaces. The same problem appeared on a draft of auto-inserting `warnOnDeprecatedApi()` calls for deprecated functions. Is there a simple ts parser that also retains format and whitespaces?

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**